conans/client/graph/graph_binaries.py

```diff
@@ -29,10 +29,11 @@ def __init__(self, conan_app, global_conf):
         self._home_folder = conan_app.cache_folder
         self._global_conf = global_conf
         self._remote_manager = conan_app.remote_manager
+        self._hook_manager = conan_app.conan_api.config.hook_manager
         # These are the nodes with pref (not including PREV) that have been evaluated
         self._evaluated = {}  # {pref: [nodes]}
         compat_folder = HomePaths(conan_app.cache_folder).compatibility_plugin_path
-        self._compatibility = BinaryCompatibility(compat_folder)
+        self._compatibility = BinaryCompatibility(compat_folder, self._hook_manager)
```

Comment:
Do we really want to execute the ``validate()`` pre/post hooks for all combinations of binary compatibility?
Can I have some examples of potential usages of the hook, and what does it mean to execute it for every binary compatibility hypothesis?

Reply:
This is still an open question

Reply:
I dont think so, I could remove it and add in the future, in case having a case.

For now, we only have the CI case to be covered.

Reply:
Then lets remove it. It is better to be minimalistic first, then add things when real use cases are requesting it.

Reply:
Done, removed on the commit [57878a1](https://github.com/conan-io/conan/pull/17856/commits/57878a1293dddff7395db9f64d041601a06b106e)

Reply:
I'd still like to see a more real use case for the feature, I haven't fully understood if it makes sense to be in the compatibility or not.

Reply:
The only real case listed right now is related to the CI: When the CI needs to skip a build, being checked in the pre_validate, but we do not want to change the recipe, by adding a new rule to raise ConanInvalidConfiguration.

About the compatibility I do not see a real usage right now, because it's related to binary compatibility mechanism, sounds be something unrelated to the validate stage.